sambacc/rados_opener.py

```diff
@@ -198,6 +251,38 @@ def write(self, s: typing.Any) -> int:
     def writelines(self, ls: typing.Iterable[typing.Any]) -> None:
         raise NotImplementedError()
 
+    def write_full(self, data: bytes) -> None:
+        """Write the object such that its contents are exactly `data`."""
+        self._ioctx.write_full(self._key, data)
+
+    def _lock_exclusive(self, name: str, cookie: str) -> None:
+        self._ioctx.lock_exclusive(
+            self._key,
+            name,
+            cookie,
+            desc=self._lock_description,
+            duration=self._lock_duration,
+        )
+
+    def _acquire_lock_exclusive(
+        self, name: str, cookie: str, *, delay: int = 1
```

Comment:
Maybe not too crucial but can be limit retries here instead of  trying indefinitely?
I think if the condition that causes the `ObjectBusy` exception persists indefinitely it could lead to a hung process that never completes.

```suggestion
        self, name: str, cookie: str, *, delay: int = 1, max_retries: int = 10
```

Reply:
Fair enough. However, I'd prefer to handle that in a follow up PR.

Reply:
Sounds good to me!